src/Pact/Types/Swagger.hs

```diff
@@ -69,6 +96,36 @@ fixedLength i =
   set maxLength (Just $ fromIntegral i) .
   set minLength (Just $ fromIntegral i)
 
+replaceOrModify :: (Eq a) => (a, a) -> (a -> a) -> a -> a
+replaceOrModify (target, replacement) modifier curr
+  | (target == curr)  = replacement
+  | otherwise         = modifier curr
```

Comment:
it'd be nice to make this a little more polymorphic so we can tell what's going on at a glance. For instance, this is a bit more descriptive of the behavior

```haskell
replaceOrModify :: Eq a => a -> a -> b -> (a -> b) -> b 
replaceOrModify t curr r k curr = bool (f t) r $ t == curr
```